### ISSUE TITLE
Add no-commented-out-code rule

### ISSUE THREAD

**Issue by vanduynslagerp** [stylelint/stylelint#1675]
Add new rule to check for commented out code.

Invalid:
```css
.foo {
  //color: red;
  color: pink;
}

//.bar {
//  color: pink;
//}

/*.baz {
  color: pink;
}*/
```

Valid:
```css
.foo {
  color: pink;
}
```

**Comment by jeddy3** [comment]
@vanduynslagerp Thanks for your suggestion. However, I don't think there's a rule way to reliably determine if a comment is commenting out code or whether it contains a comment string, other than removing the comment markers and re-parsing the code. 

As such, I don't think it's a viable candidate for an core rule. 

Anyone else have any thoughts on this, or shall I close?

**Comment by evilebottnawi** [comment]
@jeddy3 maybe `comment-no-empty` ? Disable comment without code or some words?

**Comment by evilebottnawi** [comment]
@jeddy3 as best practices empty comment in code is bad, increase size of source code and it is just an unnecessary waste, we will not check that it is there, we just forbid empty comments

**Comment by evilebottnawi** [comment]
@vanduynslagerp what do you think about rule above?

**Comment by vanduynslagerp** [comment]
Thanks so much for your consideration and quick answer !

Checking for empty comment would definitely be a nice addition to stylelint.

I still think checking for no commented out code is a good addition as well. For example, it's a really common feature in Java linting tools (http://www.sonarqube.org/commented-out-code-eradication-with-sonar/). 
But obviously it's a lot easier in Java as you can simply look for language reserved word in comments. In CSS it's a lot more complicated as selector cannot be easily identified this way.

Maybe a good approach would be to check for comments lines that start with a CSS property. This way this case would appear as error:
```css
.foo {
  //color: red;
 color: pink;
}
```

I agree that finding commented out selectors would be really difficult and might not worth the effort. So the following case probably cannot be detected:
```css
//.foo {
//}
```

Although the following one would be detected (as the second line of comment start with a CSS property):
```css
//.foo {
  //color: red;
//}
```

Now, I'm not sure if is it reliable enough.

**Comment by jeddy3** [comment]
> Now, I'm not sure if is it reliable enough.

You're more than welcome to experiment with this as a [plugin](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/plugins.md). It's the perfect candidate for one.

**Comment by vanduynslagerp** [comment]
Plugin is a good suggestion indeed !

I still think @evilebottnawi suggestion regarding `comment-no-empty` is a good idea and might be a candidate to implemented in stylelint core.

**Comment by jeddy3** [comment]
> I still think @evilebottnawi suggestion regarding comment-no-empty is a good idea and might be a candidate to implemented in stylelint core.

Yep. @evilebottnawi Can you open a new issue for that so we can discuss it?

I'll mark this issue as "plugin" and close. 

@vanduynslagerp If you do go ahead with the plugin, [this is a good ES5](https://github.com/primer/stylelint-selector-no-utility/blob/master/index.js) reference and any in [stylelint-scss](https://github.com/kristerkari/stylelint-scss/tree/master/src/rules) are good ES6 references. Good luck :)
